### PR TITLE
rg-85: Added subscriptions notifications.

### PR DESCRIPTION
**Pull request by jurok3x**
* Added email notifications on subscription.
email

### CODE REVIEW

**Review by RevolveStrm — APPROVED** [review]

File: backend/src/bundles/stripe-events/helpers/subscription-email-generator.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        const regex = new RegExp(`{${key}}`, 'g');
+        htmlContent = htmlContent.replace(regex, variables[key]);
+    }
+
+    return htmlContent;
+};
+
+const generateSubscriptionEmailPayload = ({
+    email,
+    subject = 'Subscription',
```
Comment (MaxPoplavskiy) [inline backend/src/bundles/stripe-events/helpers/subscription-email-generator.ts:18]: consider using an enum

File: backend/src/bundles/stripe-events/helpers/subscription-email-generator.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    name,
+    current_period_start,
+    current_period_end,
+}: SubscriptionDetails): SendMailOptions => {
+    const startDate = new Date(
+        current_period_start * 1000,
```
Comment (MaxPoplavskiy) [inline backend/src/bundles/stripe-events/helpers/subscription-email-generator.ts:24]: consider moving 1000 to a constant

File: backend/src/bundles/stripe-events/helpers/subscription-email-generator.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    ).toLocaleDateString();
+    const endDate = new Date(current_period_end * 1000).toLocaleDateString();
```
Comment (MaxPoplavskiy) [inline backend/src/bundles/stripe-events/helpers/subscription-email-generator.ts:26]: consider moving 1000 to a constant

**Review by MaxPoplavskiy — CHANGES_REQUESTED** [review]

File: backend/src/bundles/stripe-events/helpers/subscription-email-generator.ts (hunk 1 above)
Comment (jurok3x) [reply backend/src/bundles/stripe-events/helpers/subscription-email-generator.ts]: done

File: backend/src/bundles/stripe-events/helpers/subscription-email-generator.ts (hunk 3 above)
Comment (jurok3x) [reply backend/src/bundles/stripe-events/helpers/subscription-email-generator.ts]: done

File: backend/src/bundles/stripe-events/helpers/subscription-email-generator.ts (hunk 2 above)
Comment (jurok3x) [reply backend/src/bundles/stripe-events/helpers/subscription-email-generator.ts]: done

**Review by Icerock10 — APPROVED** [review]

File: backend/src/bundles/stripe-events/helpers/subscription-email-generator.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import { EmailPayload } from '../constants/email-payload.js';
+import { EmailSubject } from '../enums/email-subject.enum.js';
+import { type SubscriptionDetails } from '../types/types';
+
+const MILLIS_TO_SECOND = 1000;
+
+const loadAndReplaceTemplate = (variables: Record<string, string>): string => {
+    let htmlContent = EmailPayload.HTML;
+    for (const key of Object.keys(variables)) {
+        const regex = new RegExp(`{${key}}`, 'g');
```
Comment (Icerock10) [inline backend/src/bundles/stripe-events/helpers/subscription-email-generator.ts:12]: https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-resumegemm/pull/174

**Comment by jurok3x** [comment]
@Vitaliy-Kalenichenko suggest close this PR, since [#174](https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-resumegemm/pull/174)  contains all changes and much more.

**Pull request closed without merging**